src/Adapter/MSTest.TestAdapter/Extensions/ExceptionExtensions.cs

```diff
@@ -19,15 +19,30 @@ namespace Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Extensions;
 internal static class ExceptionExtensions
 {
     /// <summary>
-    /// In .NET framework, the exception thrown by the test method invocation is wrapped in a TargetInvocationException, so we
-    /// need to unwrap it to get the real exception.
+    /// TargetInvocationException and TypeInitializationException do not carry any useful information
+    /// to the user. Find the first inner exception that has useful information.
     /// </summary>
     internal static Exception GetRealException(this Exception exception)
-        => exception.GetType() == typeof(TargetInvocationException)
-            && exception.Source is "mscorlib" or "System.Private.CoreLib"
```

Comment:
@evangelink You added this code here: https://github.com/microsoft/testfx/commit/1e2abd5ff1ab79e8e423dfbaeb8a7254e017256d#diff-6cc57f94b56e0b9b613d4fc0e280124e5b9aa206c13345d3e641716af76e47dfR23-R31 

I think you are trying to make sure that the exception is thrown from methodInfo.Invoke(), but this seems unnecessary, because if there is inner exception, it will almost always have more useful info, even if user throws it themselves.  

I also cannot imagine a scenario where we would be breaking user tests in a way that makes the experience for them worse.

Reply:
I was trying to be safest possible because I only know how runtime behaves for these exceptions and not what kind of info would be available. But I agree that for these cases it seems logical to do so.

@MarcoRossignoli do you have any case in mind where this wouldn't be a good idea?